framework_tvb/tvb/adapters/visualizers/time_series_volume.py

```diff
@@ -187,24 +177,16 @@ def get_voxel_time_series(self, entity_gid, **kwargs):
                 The main part will be a vector with all the values over time from the x,y,z coordinates.
         """
 
-        ts_h5_class, ts_h5_path = self.load_h5_of_gid(entity_gid)
-
-        with ts_h5_class(ts_h5_path) as ts_h5:
-            if ts_h5_class is TimeSeriesRegionH5:
+        ts_h5 = h5.h5_file_for_gid(entity_gid)
+        with ts_h5:
+            if isinstance(ts_h5, TimeSeriesRegionH5):
                 return self._get_voxel_time_series_region(ts_h5, **kwargs)
 
-            return ts_h5.get_voxel_time_series(**kwargs)
-
-    def _get_voxel_time_series_region(self, ts_h5, x, y, z, var=0, mode=0):
-        region_mapping_volume_gid = ts_h5.region_mapping_volume.load()
-        if region_mapping_volume_gid is None:
-            raise Exception("Invalid method called for TS without Volume Mapping!")
+        return ts_h5.get_voxel_time_series(**kwargs)
```

Comment:
At this point the `ts_h5` is closed, maybe do: `data = ts_h5.get_voxel_time_series(**kwargs)` inside `with` and then `return data` here.